include/envoy/upstream/cluster_manager.h

```diff
@@ -133,6 +133,12 @@ class ClusterManager {
    */
   virtual ClusterInfoMap clusters() PURE;
 
+  /**
+   * @return absl::flat_hash_set<std::string> all current clusters names including warming and
+   * active.
+   */
+  virtual absl::flat_hash_set<std::string> allClusterNames() PURE;
```

Comment:
Lucky day, I actually need a similar change to this, but I would like to do it in a slightly different way. :)

Can you change `clusters()` above to return a struct of warming ClusterInfoMap and active ClusterInfoMap? It will be a slightly more involved change, but it's a better API change that will do what you need, do what I need, and also allow us in a later change to output warming clusters in the admin output per the TODO there (no need to do that in this PR).

Reply:
Thanks! That is the alternative start point. 

Just make it clear: `clusters()` returns both warming and active, and all other existing clusters() caller except the allClusterNames() in this PR will utilize the active map for now. 

Will do immediately after lunch.

Reply:
Yup exactly. I would just keep all others using active, and we can leave TODOs where appropriate.

Reply:
My heart is bleeding when caller doesn't need warming but clusters() blindly return both active and warming.

Is adding a Enum(active, warming, all) overkill?

Reply:
I don't think it's worth it, because I think many callers do want to use both.

Reply:
Done. no parameter is added.